crates/worker/src/partition_processor_manager.rs

```diff
@@ -289,6 +277,15 @@ impl PartitionProcessorManager {
                 Some(partition_processor_rpc) = pp_rpc_rx.next() => {
                     self.on_partition_processor_rpc(partition_processor_rpc);
                 }
+                Ok(_) = self.persisted_lsn_rx.changed() => {
+                    let (partition_id, lsn) = *self.persisted_lsn_rx.borrow_and_update();
+                    debug!(
+                        %partition_id,
+                        persisted_lsn = %lsn,
+                        "Partition persisted LSN update"
+                    );
+                    self.persisted_lsns.insert(partition_id, lsn);
```

Comment:
It doesn't look like we take any actions after observing the change. I'm not entirely sure why is this even a watch. Wouldn't it be cheaper/faster if we have an `Arc<AtomicU64>` per partition and we load it when we need to know the persistent LSN instead?

Reply:
Ended up putting this back in the `PartitionProcessorManager` where it fits much better with the singleton event listener.

Reply:
Moved this responsibility to `PartitionStoreManager`.